File: src/d400e.cpp
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2020 FRAMOS GmbH.

#include "d400e.h"
#include "smcs_cpp/CameraSDK.h"

namespace librealsense
{
    namespace d400e
    {
        heartbeat_time::heartbeat_time()
        {
            constexpr seconds default_heartbeat_time = 3;
            set(default_heartbeat_time);
        }

        heartbeat_time& heartbeat_time::get_instance()
        {
            static heartbeat_time instance;
            return instance;
        }

        void heartbeat_time::set(seconds time)
        {
            smcs::GetCameraAPI()->SetHeartbeatTime(time);
        }

        seconds heartbeat_time::get()
        {
            return smcs::GetCameraAPI()->GetHeartbeatTime();
        }
    }
}

File: wrappers/csharp/Intel.RealSense/Helpers/D400e.cs
// License: Apache 2.0. See LICENSE file in root directory.
// Copyright(c) 2020 FRAMOS GmbH.

namespace Intel.RealSense
{
    public static class D400e
    {
        public static double GetHeartbeatTime()
        {
            object err;
            return NativeMethods.rs2_d400e_get_heartbeat_time(out err);
        }

        public static void SetHeartbeatTime(double time)
        {
            object err;
            NativeMethods.rs2_d400e_set_heartbeat_time(time, out err);
        }

        public static int GetBufferCount()
        {
            object err;
            return NativeMethods.rs2_d400e_get_buffer_count(out err);
        }

        public static void SetBufferCount(int bufferCount)
        {
            object err;
            NativeMethods.rs2_d400e_set_buffer_count(bufferCount, out err);
        }

        public static int ToggleDeviceDiagnostics(string device_serial, int toggle)
        {
            object err;
            return NativeMethods.rs2_d400e_toggle_device_diagnostics(device_serial, toggle, out err);
        }
    }
}
